skbio/diversity/_block.py

```diff
@@ -0,0 +1,270 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from skbio.util._decorator import experimental
+from skbio.diversity._driver import partial_beta_diversity
+from skbio.stats.distance import DistanceMatrix
+from skbio.diversity._util import _validate_counts_matrix
+
+
+def _generate_id_blocks(ids=None, k=None, **kwargs):
+    """Generate blocks of IDs that map into a DistanceMatrix
+
+    Parameters
+    ----------
+    ids : Iterable
+        An iterable of IDs of whatever type.
+    k : int
```

Comment:
Revised, thanks! I set a default value and am explicitly checking now that `ids is not None`.